pumpkin/src/world/mod.rs

```diff
@@ -349,4 +355,39 @@ impl World {
         self.broadcast_packet_all(&CRemoveEntities::new(&[entity.entity_id.into()]))
             .await
     }
+    pub async fn set_block(&self, position: WorldPosition, block_id: BlockId) {
+        let (z_chunk, z_rem) = position.0.z.div_rem_euclid(&16);
+        let (x_chunk, x_rem) = position.0.x.div_rem_euclid(&16);
+        let chunk_coordinate = Vector2 {
+            x: x_chunk,
+            z: z_chunk,
+        };
+
+        // Since we divide by 16 remnant can never exceed u8
+        let relative = ChunkRelativeBlockCoordinates {
+            x: (x_rem as u8).into(),
+            z: (z_rem as u8).into(),
+            y: position.0.y.into(),
+        };
+
+        let (sender, mut receive) = mpsc::channel(1024);
+        {
+            let level = self.level.clone();
+            tokio::spawn(
+                async move { level.lock().await.fetch_chunks(&[chunk_coordinate], sender) },
+            );
```

Comment:
Does it make sense to spawn and await within a `fetch_chunk` or something instead of doing it externally?

We probably want to always fetch the chunk with a separate task or something.